### PR TITLE
chore: adding a few fixes to threadpool and socket

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: linux/src/threadpool_linux.c [hunk 1]
```diff
@@ -24,7 +24,8 @@
 
 #include "c_pal/threadpool.h"
 
-#define DEFAULT_TASK_ARRAY_SIZE 2048
+#define DEFAULT_TASK_ARRAY_SIZE         2048
+#define THREADPOOL_SEMEPHORE_TIMEOUT    1
```
Comment (mattdurak) [inline linux/src/threadpool_linux.c:28]: >THREADPOOL_SEMEPHORE_TIMEOUT [](http://example.com/codeflow?start=8&length=28)

THREADPOOL_SEMAPHORE_TIMEOUT_S or ...SECONDS?

Comment (mattdurak) [reply linux/src/threadpool_linux.c:28]: Is 1 second small enough? We could use `tv_nsec` instead and make it like 100ms or something

**Review by mattdurak — APPROVED** [review]
:shipit:

File: linux/src/threadpool_linux.c (hunk 1 above)
Comment (jebrando) [reply linux/src/threadpool_linux.c:28]: I can try that out, I wasn't sure how small an interval it should be.

File: linux/src/threadpool_linux.c [hunk 2]
```diff
@@ -133,8 +134,8 @@ static int threadpool_work_func(void* param)
             }
             else
             {
-                // Setup the timeout for the semaphore to 2 sec
-                ts.tv_sec += 2;
+                // Setup the timeout for the semaphore
+                ts.tv_nsec += THREADPOOL_SEMEPHORE_MS;
```
Comment (mattdurak) [inline linux/src/threadpool_linux.c:138]: >tv_nsec [](http://example.com/codeflow?start=19&length=7)

this is ns not ms so you have to multiply by 1000000

**Pull request merged**